### PR TITLE
feat(Styleguide): DecoratedStories Props and better Typography Docs

### PR DESCRIPTION
**Pull request by codecaaron**
## Notes
- Defaults to using the first found component in the `decoratedStories` args for the story component (or [0] if nothing.
- Splits Typography docs into `Heading` and `Text` respectively so stories docs can display their props.
- Moves GridForm to organism nesting and removes unused (?) stylesheet.

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9f5f0215fc06017f2dece8--codecademy-styleguide.netlify.app>https://5e9f5f0215fc06017f2dece8--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9f5f0215fc06017f2dece8>Deploy Logs</a></p>

File: packages/styleguide/stories/GridForm-story.scss [hunk 1]
```diff
@@ -1,4 +0,0 @@
-.gridFormStory {
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/GridForm-story.scss:1]: errr, sorry 😄

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9f61e7c0020c955136dfae--codecademy-styleguide.netlify.app>https://5e9f61e7c0020c955136dfae--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9f61e7c0020c955136dfae>Deploy Logs</a></p>

File: packages/styleguide/stories/Foundations/Typography/About.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Column, LayoutGrid } from '@codecademy/gamut';
+import LinkTo from '@storybook/addon-links/react';
+
+<Meta title="Foundations/Typography/About" />
+
+# Typography
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Typography/About.stories.mdx:6]: Super duper important blocking request: _where's the emoji?!_

If you need suggestions, how about ... 📜 ? 📝 ?

Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Typography/About.stories.mdx]: Rescoping to account for this important feature.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9f64eeb9a2fbe887a2a284--codecademy-styleguide.netlify.app>https://5e9f64eeb9a2fbe887a2a284--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9f64eeb9a2fbe887a2a284>Deploy Logs</a></p>

**Pull request merged**